src/importProvider.ts

```diff
@@ -0,0 +1,174 @@
+        let view = gltf.bufferViews[bufferViewIndex];
+        const offset: number = view.byteOffset === undefined ? 0 : view.byteOffset;
+        const length: number = view.byteLength;
+
+        let extension = guessFileExtension(imageBuf.mimeType);
+        let filename = targetFilename + '.' + bufferViewIndex.toString() + extension;
+        fs.writeFileSync(filename, buf.slice(offset, offset + length), 'binary');
+
+        delete imageBuf.bufferView;
+        delete imageBuf.mimeType;
+        imageBuf.uri = path.basename(filename);
+    }
+
+    // returns the shader object for the given bufferView index if the buffer view is a shader
+    function findShaderBuf(bufferViewIndex: number) : any {
+        if (gltf.shaders !== undefined) {
+            for (let shader of gltf.shaders) {
+                if (shader.bufferView == bufferViewIndex) {
+                    return shader;
+                }
+            }
+        }
+    }
+
+    // writes to the filesystem shader data from the parameters
+    function writeShaderBuf(shaderBuf: any, bufferViewIndex: number, buf: Buffer) {
+        let view = gltf.bufferViews[bufferViewIndex];
+        const offset: number = view.byteOffset === undefined ? 0 : view.byteOffset;
+        const length: number = view.byteLength;
+
+        let extension = '.glsl';
+        const GL_VERTEX_SHADER_ARB = 0x8B31;
+        const GL_FRAGMENT_SHADER_ARB = 0x8B30;
+        if (shaderBuf.type == GL_VERTEX_SHADER_ARB) {
+            extension = '.vert';
+        } else if (shaderBuf.type == GL_FRAGMENT_SHADER_ARB) {
+            extension = '.frag';
+        }
+        let filename = targetFilename + '.' + bufferViewIndex.toString() + extension;
+
+        fs.writeFileSync(filename, buf.slice(offset, offset + length), 'binary');
+
+        delete shaderBuf.bufferView;
+        delete shaderBuf.mimeType;
+        shaderBuf.uri = path.basename(filename);
+    }
+
+    // data the represents the buffers that are neither images or shaders
+    let bufferViewList:number[] = [];
+    let bufferDataList:Buffer[] = [];
+
+    function addToBinaryBuf(bufferViewIndex: number, buf: Buffer) {
+        let view = gltf.bufferViews[bufferViewIndex];
+        const offset: number = view.byteOffset === undefined ? 0 : view.byteOffset;
+        const length: number = view.byteLength;
+
+        const bufPart = buf.slice(offset, offset + length);
+
+        bufferViewList.push(bufferViewIndex);
+        bufferDataList.push(bufPart);
+    }
+
+    // go through all the buffer views and break out buffers as separate files
+    if (gltf.bufferViews !== undefined) {
+        for (let bufferViewIndex = 0; bufferViewIndex < gltf.bufferViews.length; bufferViewIndex++) {
+            let imageBuf = findImageBuf(bufferViewIndex);
+            if (imageBuf !== undefined) {
+                writeImageBuf(imageBuf, bufferViewIndex, binBuffer);
+                continue;
```

Comment:
It looks like the old `bufferView` reference is kept in the imported glTF file, but never referenced since the image was pulled out to an external file.

I tested this by importing [`Lantern.glb`](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/Lantern/glTF-Binary) from the sample model repo, and diffing the results with [`Lantern.gltf`](https://github.com/KhronosGroup/glTF-Sample-Models/blob/master/2.0/Lantern/glTF/Lantern.gltf) from the same repo.  The resulting file is almost identical, but there are four extra, un-referenced bufferView entries, with sizes that look like these images.  The `accessors` section shows all bufferView indicies have increased by 4, and in particular index `0` to `3` are never referenced.

This actually isn't a big deal, the model appears to still work fine.  But it would be nice to pull these out if it's possible.